### PR TITLE
set assigned keys when clear map

### LINKED ISSUE #663: clear Map produces incorrect patches
## 🐛 Bug Report

clear Map produce incorrect patches

## Link to repro

codebox link: https://codesandbox.io/s/immer-sandbox-forked-zz8le

### immer source code
https://github.com/immerjs/immer/blob/bf90358c1bfedf0afb6e3bb016d6e6233b96e3ac/src/plugins/mapset.ts#L112
this line clears the `assigned_`, and Patch Plugin gen patches depends on `each(state.assigned_)`

https://github.com/immerjs/immer/blob/bf90358c1bfedf0afb6e3bb016d6e6233b96e3ac/src/plugins/patches.ts#L121-L143

is this by design? or a bug.

## To Reproduce

just run the  codesanbox link  above

## Observed behavior

`changes` is a empty array.

## Expected behavior

`changes` is `[{ type: 'Remove', path: ['t2', 't'] }]`

## Environment

We only accept bug reports against the latest Immer version.

- **Immer version:**
- [x] I filed this report against the _latest_ version of Immer
- [x] Occurs with `setUseProxies(true)`
- [ ] Occurs with `setUseProxies(false)` (ES5 only)

### PR DESCRIPTION
**Pull request by LYF1999**
fix #663

### CODE REVIEW

File: src/plugins/mapset.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ import {
 	ProxyTypeSet,
 	die
 } from "../internal"
+import {each} from "../utils/common"
```
Comment (mweststrate) [inline src/plugins/mapset.ts:20]: NIT: import `each` from `internal` in the above import, as to guarantee there are no circular module deps

**Review by mweststrate — APPROVED** [review]
Looking good and thanks for adding a test as well! Left a small NIT

File: src/plugins/mapset.ts (hunk 1 above)
Comment (LYF1999) [reply src/plugins/mapset.ts]: thanks!, fixed it.

**Comment by LYF1999** [comment]
hello, is there any other problem about this PR?

**Pull request merged**